Test eos_banner

gundalow opened the pull request:
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
eos_banner


##### SUMMARY
* Integration Tests
  * Some tests are failing due to https://github.com/ansible/ansible/issues/22194 so have `ignore_errors: yes` set
  * Basic `state: present` `state: absent` including idempotent tests
  * unicode tests to ensure we encode data to/from the device correctly
* Module Docs improvements
  * Tidy up language
  * Remove `required: false`
  * Add `choices:

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ansible/ansible/22201)
<!-- Reviewable:end -->

gundalow committed:
eos_banner docs improvements

gundalow committed:
run eos_banner tests

gundalow committed:
basic eos_banner tests

mattclay commented:
CI failure due to yamllint issues:

```
2017-03-02 15:42:46 test/integration/targets/eos_banner/tests/cli/basic-login.yaml
2017-03-02 15:42:46   57:1      error    too many blank lines (1 > 0)  (empty-lines)
2017-03-02 15:42:46 
2017-03-02 15:42:46 test/integration/targets/eos_banner/tests/cli/basic-motd.yaml
2017-03-02 15:42:46   57:1      error    too many blank lines (1 > 0)  (empty-lines)
2017-03-02 15:42:46 
2017-03-02 15:42:46 test/integration/targets/eos_banner/tests/cli/basic-no-login.yaml
2017-03-02 15:42:46   49:1      error    too many blank lines (1 > 0)  (empty-lines)
```

gundalow committed:
remove trailing newlines

gundalow committed:
remove hack

gundalow committed:
Various eos_banner fixes

No unicode tests, as the device doesn't support it
Correctly set provider
result.command should be empty list if no changes

gundalow committed:
test eos_banner over eapi

gundalow committed:
revert docs changes, moved to separate PR
